switch_model/generators/core/dispatch.py

```diff
@@ -434,11 +429,31 @@ def post_solve(instance, outdir):
                  "DispatchEmissions_tCO2_per_typical_yr"]
     )
 
-    if can_plot:
-        annual_summary_plot = ggplot(
+    try:
+        import plotnine as p9
+
+        annual_summary_plot = p9.ggplot(
+                annual_summary.reset_index(),
+                p9.aes(x='period', 
+                    weight="Energy_GWh_typical_yr", 
+                    fill="factor(gen_energy_source)")
+            ) + \
+            p9.geom_bar(position="stack") + \
+            p9.scale_y_continuous(name='Energy (GWh/yr)') + \
+            p9.theme_bw()
+        save_as = os.path.join(outdir, "dispatch_annual_summary_fuel.pdf")
+        annual_summary_plot.save(filename=save_as)
+
+        annual_summary_plot = p9.ggplot(
                 annual_summary.reset_index(),
-                aes(x='period', weight="Energy_GWh_typical_yr", fill="factor(gen_tech)")
+                p9.aes(x='period', 
+                    weight="Energy_GWh_typical_yr", 
+                    fill="factor(gen_tech)")
             ) + \
-            geom_bar(position="stack") + \
-            scale_y_continuous(name='Energy (GWh/yr)') + theme_bw()
-        annual_summary_plot.save(filename=os.path.join(outdir, "dispatch_annual_summary.pdf"))
+            p9.geom_bar(position="stack") + \
+            p9.scale_y_continuous(name='Energy (GWh/yr)') + \
+            p9.theme_bw()
+        save_as = os.path.join(outdir, "dispatch_annual_summary_tech.pdf")
+        annual_summary_plot.save(filename=save_as)
+    except (ImportError, RuntimeError) as err:
+        pass
```

Comment:
Catching RuntimeError here seems a little broad, e.g., if people want the graph and can't figure out why it's not appearing. How about something more like this:

```
try:
    import plotnine as p9
except ImportError:
    pass
else:
    # all the graphing code
```

Reply:
This is addressed in https://github.com/switch-model/switch/pull/115/commits/e95abe371f52a73329c78adb8bfe6c48c579f706, along with some other improvements to streamline the code and filter out more warnings.